Documentation/contributing.rst

```diff
@@ -622,6 +622,77 @@ common unit test frameworks.
 A good use case for using ``AfterAll`` method is to remove containers or pods
 that are needed for multiple ``Its`` in the given ``Context`` or ``Describe``.
 
+JustAfterEach
+^^^^^^^^^^^^^
+
+This method will run just after each test and before ``AfterFailed`` and
+``AfterEach``. The main reason of this method is to to perform some assertions
+for a group of tests.  A good example of using a global ``JustAfterEach``
+function is for deadlock detection, which checks the Cilium logs for deadlocks
+that may have occurred in the duration of the tests.
+
+AfterFailed
+^^^^^^^^^^^
+
+This method will run before all ``AfterEach`` and after ``JustAfterEach``. This
+function is only called when the test failed.This construct is used to gather
+logs, the status of Cilium, etc, which provide data for analysis when tests
+fail.
+
+Example Test Layaout
+^^^^^^^^^^^^^^^^^^^^^
+
+Here is an example layout of how a test may be written with the aforementioned
+constructs:
+
+Test description diagram:
+::
+
+	Describe
+        BeforeAll(A)
```

Comment:
Indentation doesn't look consistent with things under `Describe` vs. those in `Context` on GitHub.